interfaces/Portalicious/src/app/domains/payment/payment.api.service.ts

```diff
@@ -7,7 +7,7 @@ import {
   PaymentStatus,
 } from '~/domains/payment/payment.model';
 
-const BASE_ENDPOINT = (projectId: Signal<number>) => [
+const BASE_ENDPOINT = (projectId: Signal<number | undefined>) => [
```

Comment:
```suggestion
const BASE_ENDPOINT = (projectId: Signal<number>) => [
```

Ideally we should never call this with an undefined projectId. If needed, change the `enabled` of the related call to this method to avoid this. I'm afraid that changing this to allow undefined could have unexpected side effects with regards to caching and query key invalidation.